### PR TITLE
SHOP-5 cleaning, dto, deliveries

### PR DESCRIPTION
**Pull request by polskikiel**

### CODE REVIEW

File: src/main/java/com/shop/dto/RegistrationDto.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    @NotBlank
+    @Pattern(regexp = "([a-zA-Z0-9]{6,32}$)")
+    private String username;
+
+    @NotBlank
+    @Pattern(regexp = "([.-^a-zA-Z0-9]{6,32}$)")
+    private String password;
+
+    @Email
```
Comment (Swistak06) [inline src/main/java/com/shop/dto/RegistrationDto.java]: Shouldn't email have @NotBlank annotation?

File: src/main/java/com/shop/dto/ProductRateDto.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import javax.validation.constraints.NotNull;
+
+public class ProductRateDto {
+
+    public ProductRateDto(@NotNull @Range(min = 1, max = 5) int mark, String comment) {
+        this.mark = mark;
+        this.comment = comment;
+    }
+
```
Comment (Wraan) [inline src/main/java/com/shop/dto/ProductRateDto.java]: Shouldn't there be @Length(max = 255) annotation before String comment?

File: src/main/java/com/shop/model/CartProduct.java [hunk 3]
```diff
@@ -13,13 +13,11 @@
...
-    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
     @JoinColumn(name = "product_id")
     private Product product;
 
-    @JsonIgnore
-    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
     @JoinColumn(name ="cart_id")
     private Cart cart;
```
Comment (Wraan) [inline src/main/java/com/shop/model/CartProduct.java]: I think there should be a fetch type eager for both of them.

Comment (Wraan) [inline src/main/java/com/shop/model/CartProduct.java]: I think there should be a fetch type eager for both of them.

File: src/main/java/com/shop/model/Delivery.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    public String getZipCode() {
+        return zipCode;
+    }
+
+    public void setZipCode(String zipCode) {
+        this.zipCode = zipCode;
+    }
+}
```
Comment (Wraan) [inline src/main/java/com/shop/model/Delivery.java]: Maybe rename this class to UserAddress or just simple Address

File: src/main/java/com/shop/model/Image.java [hunk 5]
```diff
@@ -14,12 +14,9 @@
     @Column(name = "image_id")
     private long id;
 
-    @Lob
-    @NotBlank
     private String url;
```
Comment (Wraan) [inline src/main/java/com/shop/model/Image.java:15]: It depends on host which we will be using, but some of them have links with more than 255 characters, so clob is not the worst idea.

File: src/main/java/com/shop/dto/ProductRateDto.java (hunk 2 above)
Comment (polskikiel) [reply src/main/java/com/shop/dto/ProductRateDto.java]: It should;p *when u generate constructor to early

File: src/main/java/com/shop/dto/RegistrationDto.java (hunk 1 above)
Comment (polskikiel) [reply src/main/java/com/shop/dto/RegistrationDto.java]: It should

File: src/main/java/com/shop/model/Image.java (hunk 5 above)
Comment (polskikiel) [reply src/main/java/com/shop/model/Image.java:15]: discussed

**Review by Wraan — COMMENTED** [review]
Everything good

**Review by Wraan — APPROVED** [review]

**Pull request merged**